itou/utils/emails.py

```diff
@@ -36,3 +44,111 @@ def get_email_message(to, context, subject, body, from_email=settings.DEFAULT_FR
         subject=subject_prefix + get_email_text_template(subject, context),
         body=get_email_text_template(body, context),
     )
+
+
+# Custom async email backend wrapper
+# ----------------------------------
+
+
+def _serializeEmailMessage(email_message):
+    """
+    Returns a dict with `EmailMessage` instance content serializable via Pickle (remote data sending concern).
+
+    **Important:**
+    Some important features & fields of `EmailMessage` are not "serialized":
+    * attachments
+    * special options of the messages
+
+    Just the bare minimum used by the app is kept for serialization.
+
+    This functions works in pair with `_deserializeEmailMessage`.
+    """
+    return {
+        "subject": email_message.subject,
+        "to": email_message.to,
+        "from_email": email_message.from_email,
+        "cc": email_message.cc,
+        "bcc": email_message.bcc,
+        "body": email_message.body,
+    }
+
+
+def _deserializeEmailMessage(serialized_email_message):
+    """
+        Creates a "light" version of the original `EmailMessage` passed to the email backend.
+
+        In order to be serializable, we:
+        * only get the fields actually used by the app (defined in counterpart `_serializeEmailMessage`)
+        * add a reference to the "synchronous" email backend (for convenience)
+
+        *Tip*: use non-serializable objects only when deserialization is over... (f.i. email backends)
+    """
+    return EmailMessage(connection=get_connection(backend=ASYNC_EMAIL_BACKEND), **serialized_email_message)
+
+
+@task(retries=settings.SEND_EMAIL_NB_RETRIES, retry_delay=settings.SEND_EMAIL_RETRY_DELAY)
+def _async_send_messages(serializable_email_messages):
+    """ Async email sending "delegate"
+
+        This function sends emails with the backend defined in `ASYNC_EMAIL_BACKEND`
```

Comment:
`s/function/method/` + quelques occurences en plus plus bas.

Reply:
A moins que j'ai loupé un truc? Un concept de `function` distinct des `methods` en python?

Reply:
J'ai bien utilisé "méthode" à dessein. 
Même si il n'y pas de différences au niveau Python (je crois), décorer une méthode (fonction de classe) ne fonctionnera pas avec Huey.

Reply:
Il y a une différence fondamentale entre `function` et `method` en Python comme dans beaucoup d'autres langages :

- https://docs.python.org/3/glossary.html#term-function
- https://docs.python.org/3/glossary.html#term-method

Reply:
Et une méthode à toujours l'argument `self` en premier paramètre : https://docs.python.org/3/faq/design.html#why-must-self-be-used-explicitly-in-method-definitions-and-calls

Reply:
Eh bien je suis content d'avoir posé une question idiote... j'avais oublié cette différence et dans ma tête depuis un moment toute fonction en python était automatiquement appelée une méthode lol

Au temps pour moi!